packages/fluentui/react-northstar/src/components/Menu/Menu.tsx

```diff
@@ -196,13 +198,20 @@ export const Menu = compose<'ul', MenuProps, MenuStylesProps, {}, {}>(
       setIndex(activeIndex);
     };
 
-    const handleItemOverrides = predefinedProps => ({
-      onClick: (e, itemProps) => {
+    const handleClick = React.useCallback(
+      (e, itemProps) => {
         const { index } = itemProps;
 
         setActiveIndex(e, index);
 
         _.invoke(props, 'onItemClick', e, itemProps);
+      },
+      [props, setActiveIndex],
```

Comment:
This will not work as `props` will change during each render

```tsx
function useActualProps<P>(props: P) {
  const actualProps = React.useRef<P>(props)

  React.useEffect(() => {
    actualProps.current = props
  })

  return actualProps
}

const actualProps = useActualProps(props)
```

```suggestion
        _.invoke(actualProps.current, 'onItemClick', e, itemProps);
      },
      [actualProps, setActiveIndex],
```

Reply:
There's a specific better place to me to create this `useActualProps` or should I create it here in this file itself?

Reply:
For now: let's add it here

Reply:
Done